src/Components/Forms/Company.js

```diff
@@ -33,12 +35,30 @@ class Company extends Component {
         );
     }
 
+    onButtonPress() {
+        const {name, styleGuide, primaryGoal} = this.state;
+        // if (name === '' || styleGuide === '' || primaryGoal === '') {
+        //     return alert('Must fill in all fields')
+        // }
+        return (() => {
+            this.props.updateCompanyInfo({
```

Comment:
Does this work? If not, I've just been calling the dispatch function very simply...
```
onButtonPress(){
     this.props.updateCompanyInfo(obj)
}
```
and it works. Hope I didn't throw you off with all of that talk about making anonymous functions after .then...

Reply:
no worries. The anonymous function was actually one of the few things that made sense to me haha. The post at least when i was testing wasnt working for me. I thought maybe it had to do with not having the company_id that is generated through the back end with the random number function you wrote.